lib/client.js

```diff
@@ -21,6 +21,10 @@ var defaultConnectOptions = {
   protocolId: 'MQTT',
   protocolVersion: 4,
   reconnectPeriod: 1000,
+  backOffStrategy: null,
```

Comment:
We might just support an object `BackOffStrategy`, with methods `next()`, `reset()`, `unref()`. I'm throwing in ideas, I'm not 100% sure that comparing strings is the best option.